base_layer/p2p/src/services/liveness/service.rs

```diff
@@ -267,9 +273,18 @@ where
         use LivenessRequest::*;
         match request {
             SendPing(node_id) => {
-                self.send_ping(node_id).await?;
+                let nonce = self.send_ping(node_id).await?;
                 self.state.inc_pings_sent();
-                Ok(LivenessResponse::Ok)
+                Ok(LivenessResponse::Ok(Some(vec![nonce])))
+            },
+            SendPings(node_ids, delay_between_pings) => {
+                let mut nonces = Vec::with_capacity(node_ids.len());
+                for node_id in node_ids {
+                    nonces.push(self.send_ping(node_id).await?);
+                    self.state.inc_pings_sent();
+                    time::sleep(delay_between_pings).await;
```

Comment:
this is bad
This will block all liveness requests. If you want to do this you need to move it over to a new thread so that its not blocking here.